File: solutions/csharp/resistor-color-duo/1/ResistorColorDuo.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class ResistorColorDuo
{
    private static readonly List<string> OrderedColors = new List<string>{
      "black",
      "brown",
      "red",
      "orange",
      "yellow",
      "green",
      "blue",
      "violet",
      "grey",
      "white"
    };

    public static int Value(string[] colors)
    {
        string code = "";
        foreach(string color in new List<string>(colors).Take(2))
        {
            code += OrderedColors.IndexOf(color)
                .ToString();
        }

        return int.Parse(code);
    }
}

File: solutions/python/resistor-color-duo/1/resistor_color_duo.py
RESISTOR_COLORS = [
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white",
]

def value(colors):
    first_color = RESISTOR_COLORS.index(colors[0])
    second_color = RESISTOR_COLORS.index(colors[1])
    return first_color * 10 + second_color